Sources/Common/DataModel/DataSetAttributes/FieldData.js

```diff
@@ -90,36 +100,112 @@ function vtkFieldData(publicAPI, model) {
         ) {
           if (fromId > -1 && fromId < arr.getNumberOfTuples()) {
             const tId = toId > -1 ? toId : fromId;
-            destArr.setTuple(tId, arr.getTuple(fromId));
+            destArr.insertTuple(tId, arr.getTuple(fromId));
+          } else {
+            // if fromId and not provided, just copy all (or as much possible)
+            // of arr to destArr.
+            for (let i = 0; i < arr.getNumberOfTuples(); ++i) {
+              destArr.insertTuple(i, arr.getTuple(i));
+            }
+          }
+        } else {
+          vtkErrorMacro('Unhandled case in passData');
+        }
+      }
+    });
+  };
+
+  publicAPI.interpolateData = (
+    other,
+    fromId1 = -1,
+    fromId2 = -1,
+    toId = -1,
+    t = 0.5
+  ) => {
+    other.getArrays().forEach((arr) => {
+      const copyFlag = publicAPI.getFlag(arr.getName());
+      if (
+        copyFlag !== false &&
+        !(model.doCopyAllOff && copyFlag !== true) &&
+        arr
+      ) {
+        let destArr = publicAPI.getArrayByName(arr.getName());
+        if (!destArr) {
+          if (fromId1 < 0 || fromId2 < 0 || fromId1 > arr.getNumberOfTuples()) {
+            // TBD: should this be a deep or a shallow copy?
+            publicAPI.addArray(arr);
+            other.getAttributes(arr).forEach((attrType) => {
+              publicAPI.setAttribute(arr, attrType);
+            });
+          } else {
+            const ncomps = arr.getNumberOfComponents();
+            let newSize = arr.getNumberOfValues();
+            // TODO: Is this supposed to happen?
+            const tId = toId > -1 ? toId : fromId1;
+            if (newSize <= tId * ncomps) {
+              newSize = (tId + 1) * ncomps;
+            }
+            destArr = vtkDataArray.newInstance({
+              name: arr.getName(),
+              dataType: arr.getDataType(),
+              numberOfComponents: ncomps,
+              values: macro.newTypedArray(arr.getDataType(), newSize),
+              size: 0,
+            });
+            destArr.interpolateTuple(tId, arr, fromId1, arr, fromId2, t);
+            publicAPI.addArray(destArr);
+            other.getAttributes(arr).forEach((attrType) => {
+              publicAPI.setAttribute(destArr, attrType);
+            });
+          }
+        } else if (
+          arr.getNumberOfComponents() === destArr.getNumberOfComponents()
+        ) {
+          if (fromId1 > -1 && fromId1 < arr.getNumberOfTuples()) {
+            const tId = toId > -1 ? toId : fromId1;
+            destArr.interpolateTuple(tId, arr, fromId1, arr, fromId2, t);
+            vtkWarningMacro('Unexpected case in interpolateData');
           } else {
             // if fromId and not provided, just copy all (or as much possible)
             // of arr to destArr.
+            // TODO: Create a convenient function:
+            // vtkDataArray.insertTuples(targetId, sourceArray, fromSourceId = undefined, toSourceId = undefined)
```

Comment:
let's be more generic with: `insertTuples(targetId, sourceArray)` and `getTuples()`